### PR TITLE
Fixed wrong format placeholder and nil warning on check

### PR DESCRIPTION
**Pull request by jpanda-cn**
**What type of PR is this?**

Fixed wrong format placeholder and nil warning on check

1. If this is your first time, please read our contributor guidelines: https://github.com/kubeedge/kubeedge/blob/master/CONTRIBUTING.md
2. Ensure you have added or ran the appropriate tests for your PR
3. Ignore the return value of the Close method

/kind test

### CODE REVIEW

File: edge/pkg/edgehub/common/http/http_test.go [hunk 1]
```diff
@@ -258,6 +258,10 @@ func TestBuildRequest(t *testing.T) {
 				t.Errorf("BuildRequest() error = %v, expectedError = %v", err, tt.wantErr)
 				return
 			}
+			if got == nil {
+				// will never execute, but can explicitly prevent nil warnings when checking
+				return
+			}
```
Comment (TianTianBigWang) [inline edge/pkg/edgehub/common/http/http_test.go:264]: seem that `BuildRequest` method will not return nil `got` when `err != nil`

Comment (jpanda-cn) [reply edge/pkg/edgehub/common/http/http_test.go:264]: Yes, I looked at the code for the BuildRequest and was sure that the added code was never executed, but doing so explicitly organizes editing tools and lint's nil warnings.

I can remove this part of the code if needed

**Review by fisherxu — COMMENTED** [review]
%w seems have no differece with %v? cc @gy95

**Comment by gy95** [comment]
> %w seems have no differece with %v? cc @gy95

generally, %w verb is a synonym for %v.

**Comment by TianTianBigWang** [comment]
> > %w seems have no differece with %v? cc @gy95
> 
> yes, generally, %w verb is a synonym for %v.

i got the tip `This verb can be used only in 'fmt.Errorf' calls (%w) ` in the code.
is it related?

**Comment by gy95** [comment]
> 

yes, that's it :)

File: edge/pkg/edgehub/common/http/http_test.go (hunk 1 above)
Comment (zc2638) [reply edge/pkg/edgehub/common/http/http_test.go:264]: In theory, if it is not for special circumstances, according to the specification, when `err == nil`, the returned value must be a valid value.

Comment (jpanda-cn) [reply edge/pkg/edgehub/common/http/http_test.go]: Okay, I've removed that part of the code

**Comment by gy95** [comment]
thank you @jpanda-cn  and please squash your commits to one, and sign off your commit to pass the DCO :)

**Comment by fisherxu** [comment]
> 

Thanks for clarification, and I'm wondering if it will be bug or print error if we use %w in t.Errorf? cc @TianTianBigWang @gy95

**Comment by gy95** [comment]
```
root@ecs-8f95:~/go/src# cat main.go 
package main

import(
	"fmt"
)
func main() {
	err := fmt.Errorf("failed to find key")
	fmt.Printf("%v\n", err)
	fmt.Printf("%w\n", err)
}

root@ecs-8f95:~/go/src# go run main.go 
failed to find key
%!w(*errors.errorString=&{failed to find key})
```
yes, when using %w in print format, the message is of low readability, so does `testing.Errorf`

**Comment by fisherxu** [comment]
> ```
> root@ecs-8f95:~/go/src# cat main.go 
> package main
> 
> import(
> 	"fmt"
> )
> func main() {
> 	err := fmt.Errorf("failed to find key")
> 	fmt.Printf("%v\n", err)
> 	fmt.Printf("%w\n", err)
> }
> 
> root@ecs-8f95:~/go/src# go run main.go 
> failed to find key
> %!w(*errors.errorString=&{failed to find key})
> ```
> 
> yes, when using %w in print format, the message is of low readability, so does `testing.Errorf`

Ok, we can check whether other logs should not use `%w` :)
/lgtm
/approve

**Pull request merged**